app/assets/javascripts/miq_application.js

```diff
@@ -161,64 +143,64 @@ function miqExpressionPrefill(count) {
+    $('#miq_date_1_1').prop('title', expEditor.first.title);
+    $('#miq_date_1_1').prop('alt', expEditor.first.title);
   }
   if ($('#miq_date_2_0[type=text]').length) {
-    miqPrefill($('#miq_date_2_0'), '/images/layout/expression/' + miq_val2_type + '.png');
-    $('#miq_date_2_0').prop('title', miq_val2_title);
-    $('#miq_date_2_0').prop('alt', miq_val2_title);
+    miqPrefill($('#miq_date_2_0'), '/images/layout/expression/' + expEditor.second.type + '.png');
+    $('#miq_date_2_0').prop('title', expEditor.second.title);
+    $('#miq_date_2_0').prop('alt', expEditor.second.title);
   }
   if ($('#miq_date_2_1[type=text]').length) {
-    miqPrefill($('#miq_date_2_1'), '/images/layout/expression/' + miq_val2_type + '.png');
-    $('#miq_date_2_1').prop('title', miq_val2_title);
-    $('#miq_date_2_1').prop('alt', miq_val2_title);
+    miqPrefill($('#miq_date_2_1'), '/images/layout/expression/' + expEditor.second.type + '.png');
+    $('#miq_date_2_1').prop('title', expEditor.second.title);
+    $('#miq_date_2_1').prop('alt', expEditor.second.title);
   }
-  if (typeof count == 'undefined') {
-    miq_exp_prefill_count = 0;
+  if (noPrefillCount) {
+    expEditor.prefillCount = 0;
     setTimeout(function () {
-      miqExpressionPrefill(miq_exp_prefill_count);
+      miqExpressionPrefill(expEditor, false);
     }, 200);
-  } else if (count == miq_exp_prefill_count) {
-    if (++miq_exp_prefill_count > 100) {
-      miq_exp_prefill_count = 0;
+  } else {
+    if (++expEditor.prefillCount > 100) {
+      expEditor.prefillCount = 0;
     }
     setTimeout(function () {
-      miqExpressionPrefill(miq_exp_prefill_count);
+      miqExpressionPrefill(expEditor, false);
     }, 200);
   }
 }
 
 // Prefill report editor style value text entry fields when blank
-// (written more generic for reuse, just have to build the miq_value_styles hash)
+// (written more generic for reuse, just have to build
+// the ManageIQ.reportEditor.valueStyles hash)
 function miqValueStylePrefill(count) {
   var found = false;
-  var miq_vs_prefill_count;
 
-  for (var field in miq_value_styles) {
+  for (var field in ManageIQ.reportEditor.valueStyles) {
     if ($(field).length) {
-      miqPrefill($(field), '/images/layout/expression/' + miq_value_styles[field] + '.png');
+      miqPrefill($(field), '/images/layout/expression/' + ManageIQ.reportEditor.valueStyles[field] + '.png');
       found = true;
     }
   }
   if (found) {
     if (typeof count == 'undefined') {
-      miq_vs_prefill_count = 0;
+      ManageIQ.reportEditor.prefillCount = 0;
       setTimeout(function () {
-        miqValueStylePrefill(miq_vs_prefill_count);
+        miqValueStylePrefill(ManageIQ.reportEditor.prefillCount);
       }, 200);
-    } else if (count == miq_vs_prefill_count) {
-      if (++miq_vs_prefill_count > 100) {
-        miq_vs_prefill_count = 0;
+    } else if (count == ManageIQ.reportEditor.prefillCount) {
+      if (++ManageIQ.reportEditor.prefillCount > 100) {
```

Comment:
@skateman I'm worried about readability of your suggestion. @himdel any opinion?